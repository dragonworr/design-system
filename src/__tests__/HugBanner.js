import React from 'react'
import renderer from 'react-test-renderer'
import { HugBanner, Text, theme } from '..'

describe('HugBanner', () => {
  test('renders', () => {
    const json = renderer.create(<HugBanner theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders text and icon', () => {
    const json = renderer.create(<HugBanner icon='closeLight' text={<Text.span>Hello</Text.span>} theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })
})