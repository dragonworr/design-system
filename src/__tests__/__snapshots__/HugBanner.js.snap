// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HugBanner renders 1`] = `
.c1 {
  padding: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  margin-right: 32px;
}

<div
  className="c0 c1"
>
  <svg
    className="c2"
    fill="currentcolor"
    height={20}
    mr={3}
    viewBox="0 0 20 20"
    width={20}
  >
    <path
      d="M10.002 0l2.473 7.478L20 7.64l-6 4.783L16.183 20l-6.18-4.522L3.821 20 6 12.422 0 7.64l7.528-.16z"
    />
  </svg>
</div>
`;

exports[`HugBanner renders text and icon 1`] = `
.c1 {
  padding: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  margin-right: 32px;
}

<div
  className="c0 c1"
>
  <svg
    className="c2"
    fill="currentcolor"
    height={20}
    mr={3}
    viewBox="0 0 20 20"
    width={20}
  >
    <path
      d="M10.002 0l2.473 7.478L20 7.64l-6 4.783L16.183 20l-6.18-4.522L3.821 20 6 12.422 0 7.64l7.528-.16z"
    />
  </svg>
  <span
    className=""
  >
    Hello
  </span>
</div>
`;
